File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink/20260831_075035/go/007_beatles_interview/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strings"
	"time"
)

type Relative struct {
	FirstName    string `json:"FirstName"`
	LastName     string `json:"LastName"`
	Relationship string `json:"Relationship"`
}

type Person struct {
	FirstName string     `json:"FirstName"`
	LastName  string     `json:"LastName"`
	Birthday  string     `json:"Birthday"`
	Age       int        `json:"Age"`
	Relatives []Relative `json:"Relatives"`
}

func main() {
	file, err := os.Open("input/input.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	reader := csv.NewReader(file)
	records, err := reader.ReadAll()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading CSV: %v\n", err)
		os.Exit(1)
	}

	if len(records) < 2 {
		fmt.Print("[]")
		return
	}

	header := records[0]
	var colName, colBirthday, colDied, colFather, colMother, colBrother, colSister int
	for i, h := range header {
		switch strings.TrimSpace(h) {
		case "Name":
			colName = i
		case "Birthday":
			colBirthday = i
		case "Died":
			colDied = i
		case "Father":
			colFather = i
		case "Mother":
			colMother = i
		case "Brother":
			colBrother = i
		case "Sister":
			colSister = i
		}
	}

	referenceDate := time.Date(2025, 7, 1, 0, 0, 0, 0, time.UTC)
	var people []Person

	for i := 1; i < len(records); i++ {
		record := records[i]

		// Get name parts
		fullName := strings.TrimSpace(record[colName])
		nameParts := strings.Fields(fullName)
		var firstName, lastName string
		if len(nameParts) >= 2 {
			firstName = nameParts[0]
			lastName = nameParts[len(nameParts)-1]
		} else if len(nameParts) == 1 {
			firstName = nameParts[0]
		}

		// Parse birthday
		birthdayStr := strings.TrimSpace(record[colBirthday])
		birthday, err := time.Parse("1/2/2006", birthdayStr)
		if err != nil {
			birthday, err = time.Parse("2006-01-02", birthdayStr)
			if err != nil {
				continue
			}
		}

		// Calculate age as of July 1, 2025
		age := referenceDate.Year() - birthday.Year()
		if referenceDate.Month() < birthday.Month() ||
			(referenceDate.Month() == birthday.Month() && referenceDate.Day() < birthday.Day()) {
			age--
		}

		// Build relatives list
		var relatives []Relative

		fatherName := strings.TrimSpace(record[colFather])
		if fatherName != "" && fatherName != "null" {
			fatherParts := strings.Fields(fatherName)
			fName := fatherParts[0]
			fLName := fatherParts[len(fatherParts)-1]
			relatives = append(relatives, Relative{
				FirstName:    fName,
				LastName:     fLName,
				Relationship: "Father",
			})
		}

		motherName := strings.TrimSpace(record[colMother])
		if motherName != "" && motherName != "null" {
			motherParts := strings.Fields(motherName)
			mName := motherParts[0]
			mLName := motherParts[len(motherParts)-1]
			relatives = append(relatives, Relative{
				FirstName:    mName,
				LastName:     mLName,
				Relationship: "Mother",
			})
		}

		brotherName := strings.TrimSpace(record[colBrother])
		if brotherName != "" && brotherName != "null" {
			brotherParts := strings.Fields(brotherName)
			bName := brotherParts[0]
			bLName := brotherParts[len(brotherParts)-1]
			relatives = append(relatives, Relative{
				FirstName:    bName,
				LastName:     bLName,
				Relationship: "Brother",
			})
		}

		sisterName := strings.TrimSpace(record[colSister])
		if sisterName != "" && sisterName != "null" {
			sisterParts := strings.Fields(sisterName)
			sName := sisterParts[0]
			sLName := sisterParts[len(sisterParts)-1]
			relatives = append(relatives, Relative{
				FirstName:    sName,
				LastName:     sLName,
				Relationship: "Sister",
			})
		}

		// Format birthday as YYYY-MM-DD
		birthdayFormatted := birthday.Format("2006-01-02")

		person := Person{
			FirstName: firstName,
			LastName:  lastName,
			Birthday:  birthdayFormatted,
			Age:       age,
			Relatives: relatives,
		}

		if relatives == nil {
			relatives = []Relative{}
			person.Relatives = relatives
		}

		people = append(people, person)
	}

	output, err := json.MarshalIndent(people, "", "  ")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}

	fmt.Println(string(output))
}

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink/20260831_075035/python/007_beatles_interview/solution.py
import csv
import json
from datetime import date

# Reference date for age calculation
reference_date = date(2025, 7, 1)

def calculate_age(birthday_str):
    """Calculate age as of July 1, 2025"""
    # Parse birthday in M/D/YYYY format
    parts = birthday_str.split('/')
    month = int(parts[0])
    day = int(parts[1])
    year = int(parts[2])
    
    birthday = date(year, month, day)
    
    age = reference_date.year - birthday.year
    # Check if birthday hasn't occurred yet this year
    if (reference_date.month, reference_date.day) < (birthday.month, birthday.day):
        age -= 1
    
    return age

def parse_name(full_name):
    """Parse full name into first and last name"""
    parts = full_name.strip().split()
    if len(parts) == 1:
        return parts[0], ""
    # Assume last word is last name, everything before is first name
    last_name = parts[-1]
    first_name = ' '.join(parts[:-1])
    return first_name, last_name

def main():
    results = []
    
    with open('input/input.csv', 'r') as f:
        reader = csv.DictReader(f)
        for row in reader:
            first_name, last_name = parse_name(row['Name'])
            birthday_str = row['Birthday']
            age = calculate_age(birthday_str)
            
            # Format birthday as YYYY-MM-DD
            parts = birthday_str.split('/')
            month = f"{int(parts[0]):02d}"
            day = f"{int(parts[1]):02d}"
            year = parts[2]
            birthday_formatted = f"{year}-{month}-{day}"
            
            # Build relatives list
            relatives = []
            if row['Father'] and row['Father'] != 'null':
                f_first, f_last = parse_name(row['Father'])
                relatives.append({
                    "FirstName": f_first,
                    "LastName": f_last,
                    "Relationship": "Father"
                })
            if row['Mother'] and row['Mother'] != 'null':
                m_first, m_last = parse_name(row['Mother'])
                relatives.append({
                    "FirstName": m_first,
                    "LastName": m_last,
                    "Relationship": "Mother"
                })
            if row['Brother'] and row['Brother'] != 'null':
                b_first, b_last = parse_name(row['Brother'])
                relatives.append({
                    "FirstName": b_first,
                    "LastName": b_last,
                    "Relationship": "Brother"
                })
            if row['Sister'] and row['Sister'] != 'null':
                s_first, s_last = parse_name(row['Sister'])
                relatives.append({
                    "FirstName": s_first,
                    "LastName": s_last,
                    "Relationship": "Sister"
                })
            
            results.append({
                "FirstName": first_name,
                "LastName": last_name,
                "Birthday": birthday_formatted,
                "Age": age,
                "Relatives": relatives
            })
    
    print(json.dumps(results, indent=2))

if __name__ == '__main__':
    main()